### PR TITLE
Snyder - LB124.lc_r_unmgd_yh_glu

### PR DESCRIPTION
**Pull request by abigailsnyder**
#642

### CODE REVIEW

File: R/zchunk_L124.nonco2_unmgd_R_S_T_Y.R [hunk 1]
```diff
@@ -22,8 +22,8 @@ module_emissions_L124.nonco2_unmgd_R_S_T_Y <- function(command, ...) {
   if(command == driver.DECLARE_INPUTS) {
     return(c(FILE = "common/iso_GCAM_regID",
              FILE = "emissions/EDGAR/EDGAR_sector",
-             FILE = "temp-data-inject/L124.LC_bm2_R_Grass_Yh_GLU_adj",
-             FILE = "temp-data-inject/L124.LC_bm2_R_UnMgdFor_Yh_GLU_adj",
+             "L124.LC_bm2_R_Grass_Yh_GLU_adj",
+             "L124.LC_bm2_R_UnMgdFor_Yh_GLU_adj",
```
Comment (bpbond) [inline R/zchunk_L124.nonco2_unmgd_R_S_T_Y.R]: Did you remove corresponding files from `temp-data-inject/`?

File: R/zchunk_LB124.LC_R_UnMgd_Yh_GLU.R [hunk 2]
```diff
@@ -35,73 +46,173 @@ module_aglu_LB124.LC_R_UnMgd_Yh_GLU_DISABLED <- function(command, ...) {
...
+      # Then, Calculate adjustment ratio = (Total Unmgd Land - Extra Cropland) / Total Unmgd Land
+      # to get adjusted total unmanaged land, keeping NAs for later processing:
+      left_join(select(L122.LC_bm2_R_ExtraCropLand_Yh_GLU, -Land_Type), by = c("GCAM_region_ID", "GLU", "year")) %>%
+      rename(ExtraCropland = value) %>%
+      mutate(adjustmentRatio = (TotUnmgdLand - ExtraCropland) / TotUnmgdLand) %>%
+      replace_na(list(adjustmentRatio = 1)) %>%
+      select(-TotUnmgdLand, -ExtraCropland, -Land_Type) ->
+      L124.LC_UnMgdAdj_R_Yh_GLU
+
+    #Check that enough unmanaged land is available for the cropland expansion in all regions/GLUs
```
Comment (bpbond) [inline R/zchunk_LB124.LC_R_UnMgd_Yh_GLU.R]: `# Check`

File: R/zchunk_LB124.LC_R_UnMgd_Yh_GLU.R [hunk 3]
```diff
@@ -35,73 +46,173 @@ module_aglu_LB124.LC_R_UnMgd_Yh_GLU_DISABLED <- function(command, ...) {
...
+    if( any( L124.LC_UnMgdAdj_R_Yh_GLU$adjustmentRatio < 0 ) ) {
+      stop( "Increase in cropland exceeds available unmanaged land")
+    }
+
+
+    # Apply the adjusment ratio to the different land types
+    L124.LC_bm2_R_LTunmgd_Yh_GLU %>%
+      left_join_error_no_match( L124.LC_UnMgdAdj_R_Yh_GLU, by = c("GCAM_region_ID", "GLU", "year")) %>%
+      mutate(value = value * adjustmentRatio) %>%
+      select(- adjustmentRatio) ->
```
Comment (bpbond) [inline R/zchunk_LB124.LC_R_UnMgd_Yh_GLU.R]: `-adjustmentRatio`

File: R/zchunk_LB124.LC_R_UnMgd_Yh_GLU.R [hunk 4]
```diff
@@ -35,73 +46,173 @@ module_aglu_LB124.LC_R_UnMgd_Yh_GLU_DISABLED <- function(command, ...) {
...
+      add_comments("the total amount of unmanaged land in each region-GLU-year. This is then used with") %>%
+      add_comments("the extra cropland calculated in L122.LC_bm2_R_ExtraCropLand_Yh_GLU to create an adjustment") %>%
+      add_comments("factor for each region-GLU-year. This adjustment factor is applied to each unmanaged land") %>%
+      add_comments("type in each region-GLU-year so that the extra cropland is deducted from each unmanaged land") %>%
+      add_comments("type according to its relative share.") %>%
       add_legacy_name("L124.LC_bm2_R_Grass_Yh_GLU_adj") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
+      add_precursors("L120.LC_bm2_R_LT_Yh_GLU",
```
Comment (bpbond) [inline R/zchunk_LB124.LC_R_UnMgd_Yh_GLU.R]: Here and below, don't forget about `same_precursors_as` for brevity.

**Comment by abigailsnyder** [comment]
@bpbond  do you have any idea why the build is failing? I checked as carefully as I could on my machine that it passed check package and test package at different stages.

**Comment by bpbond** [comment]
Yes see #656 -

**Comment by abigailsnyder** [comment]
cool, I was wondering if it related to that or if I had a genuine mistake on my part

File: R/zchunk_LB124.LC_R_UnMgd_Yh_GLU.R [hunk 5]
```diff
@@ -35,73 +46,164 @@ module_aglu_LB124.LC_R_UnMgd_Yh_GLU_DISABLED <- function(command, ...) {
...
     #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # Calculate the adjustment ratio for this deduction.
+    # adjustment ratio = (Total Unmgd Land - Extra Cropland) / Total Unmgd Land
+    # First, calculate the total amount of unmanaged land:
+    L124.LC_bm2_R_LTunmgd_Yh_GLU%>%
```
Comment (rynacui) [inline R/zchunk_LB124.LC_R_UnMgd_Yh_GLU.R]: space before `%>%`

File: R/zchunk_LB124.LC_R_UnMgd_Yh_GLU.R [hunk 6]
```diff
@@ -35,73 +46,164 @@ module_aglu_LB124.LC_R_UnMgd_Yh_GLU_DISABLED <- function(command, ...) {
...
+
+    # Check that enough unmanaged land is available for the cropland expansion in all regions/GLUs
+    if( any( L124.LC_UnMgdAdj_R_Yh_GLU$adjustmentRatio < 0 ) ) {
+      stop( "Increase in cropland exceeds available unmanaged land")
+    }
+
+
+    # Apply the adjusment ratio to the different land types
+    L124.LC_bm2_R_LTunmgd_Yh_GLU %>%
+      left_join_error_no_match( L124.LC_UnMgdAdj_R_Yh_GLU, by = c("GCAM_region_ID", "GLU", "year")) %>%
```
Comment (rynacui) [inline R/zchunk_LB124.LC_R_UnMgd_Yh_GLU.R]: no space after `(`

**Comment by bpbond** [comment]
Travis is down because of #656 but merging anyway.

**Pull request merged**